modules/core/src/matrix_wrap.cpp

```diff
@@ -15,7 +15,7 @@ namespace cv {
 Mat _InputArray::getMat_(int i) const
 {
     int k = kind();
-    int accessFlags = flags & ACCESS_MASK;
+    AccessFlag accessFlags = flags & ACCESS_MASK;
 
```

Comment:
What is about this?
```
AccessFlag operator & (const int a, const AccessFlag& b)
```

Reply:
This also would work, but I am not sure if `int` is the only type utilized with bitwise operations (how about `long`?), and I do not think we want to provide all the variants. On the other hand, the template-based solution referenced above is just doing that with minimal effort, I guess.

Reply:
`int`, `long` and other helper operators can be placed under `CV_FLAGS_ENUM(enumType)` macro:
```
enum AccessFlag { ... };
CV_FLAGS_ENUM(AccessFlag);  // define helper operators here
```

It can be added for each bit-set/flags enum types.

Reply:
@alalek I switched to the macros. Do you still have any particular issues?